primo2/inference/marginal.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Feb 23 13:37:05 2017
+
+@author: jpoeppel
+"""
+
+import numpy as np
+
+class Marginal(object):
+    
+    """
+        A class representing the inference results. This class holds the
+        (joint) probability distribution after performing inference.
+        
+        TODO: Consider adding potentially used evidence and it's probability
+        as well.
+    """
+    
+    def __init__(self):
+        self.variables = []
+        self.values = {}
+        self.probabilities = 0
+        
+    @classmethod
+    def from_factor(cls, factor):
+        """
+            Creates a marginal from a factor. This method should only be used
+            internally as a factor does not make any guarantees about the kind
+            of potential it contains, thus calling it with factors not containing
+            marginal probabilities will result in invalid marginals!
+            
+            Parameters
+            ----------
+            factor: Factor
+                The factor whose potential is used to construct the marginal.
+                
+            Returns
+            --------
+                Marginal
+                The created marginal representing the (joint) posterior
+                marginal over all the variables in the factor.
+        """
+        res = cls()
+        res.variables = factor.variableOrder
+        res.values = dict(factor.values)
+        res.probabilities = factor.potentials.copy()
+        return res
+    
+    def get_probabilities(self, variables=None, returnDict=False):
+        """
+            Returns the probabilities for the specified variable(s), if specified,
+            either as a compact numpy array (default), ordered according to the 
+            variable and corresponding value orders in self.variables and 
+            self.values, or as a dictionary.
+            
+            If variables is not specified it will return the probabilities for
+            all included variables in the specfied form.
+            
+            Parameter
+            ---------
+            variables: dict, string, optional.
+                Dictionary containing the desired variable names as keys and
+                either an instantiation or a list of instantiations of interest 
+                as values. For a marginal containing the binary variables A and B, 
+                get_probabilities({"A":"True"}) and get_probabilities({"A":["True"]})
+                will return the probabilties P(A=True, B=True) and 
+                P(A=True, B=False). Whereas get_probabilities({"A":"True", "B":"False"})
+                will only return P(A=True,B=False).
+                
+            returnDict: Boolean, optional (default: False)
+                Specifies if the probabilities should be returned as a dictionary
+                of the form {variable: probabilities} (if set to true) or as
```

Comment:
True, at least for the general case. For the simple case (which was the only one I considered writing this as I see now) where I only have 1 variable, I directly return the inner dictionairy, as the outer one is kind of pointless. But I'll change the docstring to be more accurate on this.